soccer/gameplay/skills/dribble.py

```diff
@@ -0,0 +1,125 @@
+import single_robot_composite_behavior
+import behavior
+import robocup
+import role_assignment
+import main
+import constants
+import time
+import enum
+import skills.aim
+import skills.capture
+
+
+## Behavior that moves the ball to a specified location
+class Dribble(single_robot_composite_behavior.SingleRobotCompositeBehavior):
+    class State(enum.Enum):
+        capture = 1
+        aim = 2
+        drive = 3
+
+    def __init__(self, pos=None, vel=0):
+        super().__init__(continuous=False)
+
+        for state in Dribble.State:
+            self.add_state(state, behavior.Behavior.State.running)
+
+        self.threshold = 0.01  #default value matches the required accuracy for a placement command
+        self.pos = pos
+        self.vel = vel
+
+        self.add_transition(behavior.Behavior.State.start,
+                            Dribble.State.capture, lambda: True, 'immediately')
+
+        self.add_transition(
+            Dribble.State.capture, Dribble.State.aim,
+            lambda: self.subbehavior_with_name('capture').state == behavior.Behavior.State.completed,
+            'done capturing')
+
+        #Put the ball between the robot and the target
+        self.add_transition(
+            Dribble.State.aim, Dribble.State.drive,
+            lambda: self.subbehavior_with_name('aim').state == skills.aim.Aim.State.aimed,
+            'done aiming')
+
+        self.add_transition(
+            Dribble.State.aim, Dribble.State.capture,
+            lambda: self.subbehavior_with_name("aim").state == behavior.Behavior.State.failed,
+            'fumbled')
+
+        self.add_transition(Dribble.State.drive, Dribble.State.capture,
+                            lambda: self.fumbled() or self.robot.vel.mag() > 1,
```

Comment:
I'm not sure exactly why this transition happens if the robot's velocity is >1? If you want to keep the robot moving slowly, you might want to add a max speed  parameter to dribble instead?